### PR TITLE
Feature/iep 634 refactor insolvency api post routes

### PR DESCRIPTION
**Pull request by adeharrison-ch**

### CODE REVIEW

File: utils/common_validation_test.go [hunk 1]
```diff
@@ -24,53 +24,141 @@ func prepareForTest() (*http.Request, *httptest.ResponseRecorder) {
...
-		So(actual, ShouldBeFalse)
+		So(transactionId_, ShouldEqual, "")
+		So(valid, ShouldBeFalse)
+		So(httpStatusCode, ShouldEqual, http.StatusBadRequest)
 	})
 
 	Convey("Passes validation when transaction ID exists", t, func() {
 		var req, res = prepareForTest()
-		_, actual, _ := helperService.HandleTransactionIdExistsValidation(res, req, "1234567")
+		transactionId_, valid, httpStatusCode := helperService.HandleTransactionIdExistsValidation(res, req, "1234567")
```
Comment (Jideboris) [inline utils/common_validation_test.go:39]: any reason why this is named like this.I thought transactionId should suffice

File: utils/common_validation_test.go [hunk 2]
```diff
@@ -24,53 +24,141 @@ func prepareForTest() (*http.Request, *httptest.ResponseRecorder) {
 
 func TestUnitHandleRequestValidation(t *testing.T) {
 	helperService := NewHelperService()
+
 	Convey("Fails validation when transaction ID does exists", t, func() {
 		var req, res = prepareForTest()
-		_, actual, _ := helperService.HandleTransactionIdExistsValidation(res, req, "")
+		transactionId_, valid, httpStatusCode := helperService.HandleTransactionIdExistsValidation(res, req, "")
```
Comment (Jideboris) [inline utils/common_validation_test.go:30]: any reason why this is named like this.I thought transactionId should suffice

**Review by Jideboris — APPROVED** [review]

File: utils/common_validation_test.go (hunk 1 above)
Comment (adeharrison-ch) [reply utils/common_validation_test.go]: done

File: utils/common_validation_test.go (hunk 2 above)
Comment (adeharrison-ch) [reply utils/common_validation_test.go]: done

File: utils/common_validation.go [hunk 3]
```diff
@@ -38,8 +70,38 @@ func (*helperService) HandleBodyDecodedValidation(w http.ResponseWriter, req *ht
...
+		m := models.NewMessageResponse(errs)
+		WriteJSONWithStatus(w, req, m, http.StatusBadRequest)
+		return false, http.StatusBadRequest
+	}
+	return true, http.StatusOK
+}
+
+// HandleAttachmentValidation implements HelperService
+func (*helperService) HandleAttachmentValidation(w http.ResponseWriter, req *http.Request, transactionID string, attachment models.AttachmentResourceDao, err error) (bool, int) {
+	if attachment == (models.AttachmentResourceDao{}) {
```
[the remark this answers is not in the document]
Comment (adeharrison-ch) [reply utils/common_validation.go:87]: agreed to leave as original logic

File: handlers/soa_resource.go [hunk 4]
```diff
@@ -79,36 +70,37 @@ func HandleCreateStatementOfAffairs(svc dao.Service) http.Handler {
...
 		}
 
-		attachment, err := svc.GetAttachmentFromInsolvencyResource(transactionID, statementDao.Attachments[0])
-
 		// Validate if supplied attachment matches attachments associated with supplied transactionID in mongo db
-		if attachment == (models.AttachmentResourceDao{}) {
-			log.ErrorR(req, fmt.Errorf("failed to get attachment from insolvency resource in db for transaction [%s] with attachment id of [%s]: %v", transactionID, statementDao.Attachments[0], err))
-			m := models.NewMessageResponse("attachment not found on transaction")
-			utils.WriteJSONWithStatus(w, req, m, http.StatusInternalServerError)
+		attachment, err := svc.GetAttachmentFromInsolvencyResource(transactionID, statementDao.Attachments[0])
```
[the remark this answers is not in the document]
Comment (adeharrison-ch) [reply handlers/soa_resource.go:63]: agreed to leave as original logic

File: handlers/resolution_resource.go [hunk 5]
```diff
@@ -90,45 +71,44 @@ func HandleCreateResolution(svc dao.Service, helperService utils.HelperService)
...
 		}
 
-		attachment, err := svc.GetAttachmentFromInsolvencyResource(transactionID, resolutionDao.Attachments[0])
-
 		// Validate if supplied attachment matches attachments associated with supplied transactionID in mongo db
-		if attachment == (models.AttachmentResourceDao{}) {
-			log.ErrorR(req, fmt.Errorf("failed to get attachment from insolvency resource in db for transaction [%s] with attachment id of [%s]: %v", transactionID, resolutionDao.Attachments[0], err))
-			m := models.NewMessageResponse("attachment not found on transaction")
-			utils.WriteJSONWithStatus(w, req, m, http.StatusInternalServerError)
+		attachment, err := svc.GetAttachmentFromInsolvencyResource(transactionID, resolutionDao.Attachments[0])
```
[the remark this answers is not in the document]
Comment (adeharrison-ch) [reply handlers/resolution_resource.go:71]: agreed to leave as original logic

File: handlers/progress_report_resource.go [hunk 6]
```diff
@@ -19,57 +17,37 @@ import (
...
-		log.InfoR(req, fmt.Sprintf("start POST request for submit progress report with transaction id: %s", transactionID))
-
-		isTransactionClosed, err, httpStatus := service.CheckIfTransactionClosed(transactionID, req)
-
-		_, validTransactionNotClosed, httpStatusCodes := helperService.HandleTransactionNotClosedValidation(w, req, transactionID, isTransactionClosed, err, httpStatus)
-
-		if !validTransactionNotClosed {
-			http.Error(w, "Transaction closed", httpStatusCodes)
+		// Check transaction is valid
+		transactionID, isValidTransaction, httpStatusCode, errMessage := utils.HandleTransactionValidation(helperService, req, w, "progress report", service.CheckIfTransactionClosed)
```
[the remark this answers is not in the document]
Comment (Jideboris) [reply handlers/progress_report_resource.go]: Discussed

File: handlers/resolution_resource.go [hunk 7]
```diff
@@ -18,63 +18,33 @@ import (
...
-			utils.WriteJSONWithStatus(w, req, m, httpStatus)
-			return
-		}
-
-		if isTransactionClosed {
-			log.ErrorR(req, fmt.Errorf("transaction [%v] is already closed and cannot be updated", transactionID))
-			m := models.NewMessageResponse(fmt.Sprintf("transaction [%v] is already closed and cannot be updated", transactionID))
-			utils.WriteJSONWithStatus(w, req, m, httpStatus)
+		// Check transaction is valid
+		transactionID, isValidTransaction, httpStatusCode, errMessage := utils.HandleTransactionValidation(helperService, req, w, "resolution", service.CheckIfTransactionClosed)
```
[the remark this answers is not in the document]
Comment (Jideboris) [reply handlers/resolution_resource.go]: Discussed

**Review by Jideboris — APPROVED** [review]

**Pull request merged**